gobblin-core/src/main/java/gobblin/writer/RetryWriter.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.writer;
+
+import gobblin.configuration.State;
+import gobblin.instrumented.Instrumented;
+import gobblin.metrics.GobblinMetrics;
+
+import java.io.IOException;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.Meter;
+import com.github.rholder.retry.Attempt;
+import com.github.rholder.retry.RetryListener;
+import com.github.rholder.retry.Retryer;
+import com.github.rholder.retry.RetryerBuilder;
+import com.github.rholder.retry.StopStrategies;
+import com.github.rholder.retry.WaitStrategies;
+import com.google.common.base.Optional;
+import com.google.common.base.Predicate;
+
+/**
+ * Retry writer follows decorator pattern that retries on inner writer's failure.
+ * @param <D>
+ */
+public class RetryWriter<D> implements DataWriter<D> {
+  private static final Logger LOG = LoggerFactory.getLogger(RetryWriter.class);
+  public static final String FAILED_RETRY_WRITES_METER = "gobblin.writer.failed.retry.writes";
```

Comment:
Please do not use "." as replacements of spaces. You can use "_" as you did for `CONVERTER_AVRO_REST_ENTRY_RESOURCE_KEY`.